### PR TITLE
EM patch 1

### PR DESCRIPTION
**Pull request by emmoramor**

### CODE REVIEW

**Review by Arachnophobian — COMMENTED** [review]
Hello,

As stated in class, I don't know much but I looked through for anywhere that I might offer some feedback.

Question #2 - For (b) and (c), when you say "CNF" next to the answer, do you mean to say that the original formula is in CNF? I might have misunderstood, but I thought that there simply wasn't a CNF form for tuatologies because it would have to have both conjunction and disjunction. If this is right, then when you wrote (A v ~B), does this fail to be CNF because it's only a disjunction?

Separate from the above, what made you choose ~B and not ~A? Does (A v ~B) still count as a tuatology?

Question #5 - I'm not sure I understand T-Boxes and A-Boxes thoroughly, but I wanted to know if when you wrote "B that says that [w] is a brother of [w]" this would combines with the A-Box to mean "John is a brother of John"? I cannot read the DL answers you gave for #4,A), so ignore this if I'm incorrect.

Question #6 - In (a) you wrote "there is or exists some x such that for all y..." but I believe this is incorrect; I'm not sure that the ∃ should be read to conjunct "is" and "exists", given that conjuncts can be read as "either one or neither". It's a minor thing, of course, and I didn't see that you wrote it that way anywhere else.

Question #7 - I think this graphing method is very well done and just wanted to give kudos. I know it probably seems incredibly simple, but for me I was like "Oh shit... that makes total sense". XD

**Comment by Jiemingy** [comment]
Hi Emanuel,
I just learned something from other people in the class, and use it for the review. Hope it helps. 
Q2. a CNF of tautology is any form of p v ~p, so your answer to (b) seems wrong. It could be any one of A v ~ A, B v ~ B or C v ~C.
Q.3 (c) there should be a negation before (x=y)
(d) your formula seems not to exclude there is another child: it just say that x is a child of y. 
(e) you may not need the first ∀z. 
In addition, just learned from Karl that when we define B(x,y), both x and y should be free in the definition - I am not sure, but it makes sense.  
Q.5 I wonder if you need to define Nephew in the T-box, since it appears in the A-box.

**Review by JaronJCheung — COMMENTED** [review]
Dear @emmoramor, 

Thank you for inviting me into the process of reviewing your work! I have reviewed your work and below are some suggestions, observations, and some sharing of material. 

Q3: Your logic sentence constructions seem to make sense as I read them and perhaps there are multiple ways to write the same sentence. However, I wrote my sentences in different ways. I will share my answers so that you can compare them.

(a) B(x,y) that says that x is a brother of y:

B(xy) = ∃z(P(z,x) ∧ P(z,y)) ∧ ~F(x)

(b) A(x,y) that says that x is an aunt of y

A(xy) = ∃z∃u(P(u,x) ∧ P(u,z) ∧ P(z,y) ∧ F(x))

(c) C(x,y) that says that x and y are cousins

C(xy) = ∃u∃w∃z(P(u,w) ∧ P(u,z) ∧ P(w,x) ∧ P(z,y) ∧ ~(z=u)) ∧ ~(x=y)

(d) O(x) that says that x is an only child

O(x) = ∃z(P(z,x) ∧ ∀y(P(z,y) → x=y))

(e) T(x) that says that x has exactly two brothers

T(x) = ∃w∃y∃z(P(w,x) ∧ P(w,y) ∧ P(w,z) ∧ ~(x=y) ∧ ~(x=z) ∧ ~(y=z) ∧ ~F(y) ∧ ~F(z) ∧ ∀u(P(w,u) → u=x ∨ u=y ∨ u=z) ∧ ~F(u))

Q6: It might be good to explain what the relationship between each quantifier is (no stress, however, if you think it is not necessary). Below is how I went about it. Please feel free to glean (or not) whatever is helpful!

(a) ∃x∀y and ∀x∃y

∃x∀y = "There exists an x such that for every y" or possibly "Something is identical with everything." This can be interpreted as a statement saying that there is at least one value of x for which the statement "for every y" is true. In other words, it asserts the existence of x in relation to all possible values of y.

∀x∃y = "For all x, there exists a y" or possibly "Everything is identical with something" This can be interpreted as a statement saying that for every possible value of x, there is at least one corresponding value of y. In other words, it asserts the existence of y in relation to all possible values of x.

(b) ∃x∀y∃z and ∀x∃y∀z

∃x∀y∃z = "There exists an x such that for every y, there exists a z." This can be interpreted as a statement saying that for some value of x, we can find a corresponding value of z for every possible value of y. It states the existence of x and z in relation to all possible values of y.

∀x∃y∀z = "For all x, there exists a y such that for all z." This can be interpreted as a statement saying that for all possible values of x, we can find a corresponding value of y that holds true for all values of z. In other words, it asserts the existence of y in relation to all possible values of x and z.

(c) ∀x∃y∀z∃w and ∃x∀y∃z∀w

∀x∃y∀z∃w = "For all x, there exists a y such that for all z, there exists a w." This can be interpreted as a statement saying that for all possible values of x, we can find a corresponding value of y, and for all possible values of z, we can find a corresponding value of w. In other words, it asserts the existence of y and w in relation to all possible values of x and z.

∃x∀y∃z∀w = "There exists an x such that for all y, there exists a z for all w." This can be interpeted as a statement saying that for some value x, we can find a corresponding value y, and for every y, there is some value z that holds for all of w.

Q7: It seems that your universe does not satisfy either sentence. 

  0 ----> 1 ----> 2
    `--------------^

   U: {0,1,2}
   R: {(0,1), (0,2), (1,2)}

For example, on ∀x∃y∀z (R(x,y) ∧ R(x,z) ∧ R(y,z)), every point does not "R" something (i.e. 2 is not R'ing anything, which makes it false in this universe. 

On ∃x∀y∃z(R(x,y) ∧ R(x,z) ∧ R(y,z)), every point does not "R" everything.

To make it easier, I minimized the universe to two points and I drew a diagram that could satisfy sentence one and falsified the second by showing that everything is not "R'ing" everything.

e.g. 

U: {0, 1}

R: {(0. 0), (0, 1), {1, 1)}

Q10: Your response is correct! If I may suggest adding one more caveat, the tree proofs rely only on indirect derivation, i.e. finding a contradiction, to prove its conclusion, where as natural deduction has multiple inference rules and derivations that can be used to prove a conclusion.

I hope this all helps. Let me know if you have any follow up questions. 

Best, 
Jaron

File: Project1_tr [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+  
+  Already a tautology; CNF  
+  A v ~A
+
+
+3. Let V be the vocabulary of first-order logic consisting of a binary relation P and a unary relation F. Interpret P(x,y) as “x is a parent of y” and F(x) as “x is female.” Where possible define the following formulas in this vocabulary; where not possible, explain why:
+
+(a)  B(x,y) that says that x is a brother of y 
+  
+  Def.   ∃x∃y∃z(P(z,y)^P(z,x)^¬Fx^¬(y=x))
```
Comment (johnbeve) [inline Project1_tr:106]: The variables for B(x,y) should be under universal scope. As written, this definition of 'brother of' is satisfied by one pair, but a definition of 'brother of' should hold for all pairs that meet the conditions.

File: Project1_tr [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+
+(b)  A(x,y) that says that x is an aunt of y  
+  
+  Def.   ∃w∃x∃y∃z(P(w,z)^P(w,x)^P(z,y)^¬(z=x)^F(x))
```
Comment (johnbeve) [inline Project1_tr:110]: Counterexample: Aunt by marriage

File: Project1_tr [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+
+(c)  C(x,y) that says that x and y are cousins   
+  
+  Def.  ∃v∃w∃x∃y∃z(P(v,w)^P(v,z)^P(w,x)^P(z,y)^¬(w=z)^¬(x=y))
```
Comment (johnbeve) [inline Project1_tr:114]: Counterexample: Second cousins

File: Project1_tr [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+  R: {(0,1),(0,2),(1,2)}
+  
+  
+  
+8. Using an online tableau proof generator - such as the one found here https://www.umsu.de/trees/ - provide tree proofs of the following entailments, 
+which are known as the De Morgan's laws:
+
+(a) ∀x∀y(¬(Px ∧ Qx) → (¬Px ∨ ¬Qx))
+
+<img src="blob:chrome-untrusted://media-app/55dfa8c9-4008-4723-9c14-e4c8b00119b0" alt="Project1_8a.png"/>
```
Comment (johnbeve) [inline Project1_tr:234]: What justifies moving from the negated universal in line 1 to the negated universal in line 2?

File: Project1_tr [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+1,10	          (14)	         ¬Px ∧ Px	                           13 e.f.q.
+1	              (15)	         ¬Px ∧ Px	                           5,6,9,10,14∨B
+                (16)	         ¬¬((¬Px ∧ ¬Qx) → ¬(Px ∨ Qx))	      1,15RAA
+                (17)	         (¬Px ∧ ¬Qx) → ¬(Px ∨ Qx)	          16DN
+
+
+
+10. Compare and contrast the proofs provided for (a) in your answers to questions 8 and 9. Explain the different assumptions, strategies, etc. exhibited in tree proofs vs natural deduction proofs.
+
+The main difference or contrast between the tree-proofs in question 9 and the natural deduction proofs in question 10 is that the tree proofs omit the rules/justifications that signal each step of the proof. The natural deductions, by contrast, do not. In fact, the proof generator I used for question 10 annotated rather heavily.
```
Comment (johnbeve) [inline Project1_tr:348]: Is there anything different about the structures of the respective proofs generated?

**Review by johnbeve — CHANGES_REQUESTED** [review]

File: Project1_tr (hunk 4 above)
Comment (emmoramor) [reply Project1_tr:244]: The move from the negated universal in line 1 to the negated universal in line 2 is justified by the rule of negation in tableau proof. The rule of negation says that if a formula ¬A is on an open branch, then you can add A to that branch with a negation sign in front of it. This allows us to assume the negation of a formula and derive a contradiction, which means that the formula itself must be true. In this case, we assume the negation of the formula in line 2, and derive a contradiction by instantiating it with the same variable, which allows us to close the branch and infer the truth of the original formula.

**Comment by emmoramor** [comment]
Dear Prof. Beverley,

I made the changes you requested and marked the conversations as resolved; I also answered your question re 8a in the conversation, before marking it resolved.

Best,
EM

**Pull request closed without merging**